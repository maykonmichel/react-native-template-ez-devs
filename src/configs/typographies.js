import { keys } from 'lodash';

const create = (fontSize, lineHeight, fontFamily) => ({
  fontSize,
  lineHeight,
  fontFamily
});

const typographies = {
  display5: create(112, 128),
  display4: create(80, 100),
  display3: create(56, 64),
  display2: create(45, 52),
  display1: create(34, 40),
  title1: create(28, 34),
  title2: create(22, 28),
  title3: create(20, 25),
  headline: create(18, 23),
  body: create(17, 22),
  callout: create(16, 21),
  subhead: create(15, 20),
  footnote: create(13, 18),
  caption1: create(12, 16),
  caption2: create(11, 13)
};

export default typographies;

export const typographiesKeys = keys(typographies);
