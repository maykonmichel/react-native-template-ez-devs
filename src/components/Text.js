import React, { useMemo } from 'react';
import { StyleSheet, Text as RNText } from 'react-native';
import PropTypes from 'prop-types';

import typographies, { typographiesKeys } from '../configs/typographies';

const Text = ({ children, typography, style, ...attributes }) => {
  const typographyStyle = useMemo(() => typographies[typography], [typography]);

  return (
    <RNText style={StyleSheet.flatten([typographyStyle, style])} {...attributes}>
      {children}
    </RNText>
  );
};

Text.propTypes = {
  ...RNText.propTypes,
  style: RNText.propTypes.style,
  typography: PropTypes.oneOf(typographiesKeys)
};

Text.defaultProps = {
  style: {},
  typography: 'headline'
};

export default Text;
