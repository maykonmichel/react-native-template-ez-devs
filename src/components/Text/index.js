import React from 'react';
import { StyleSheet, Text as RNText } from 'react-native';
import PropTypes from 'prop-types';

import typographies, { typographiesKeys } from '../../configs/typographies';

Text.propTypes = {
  ...RNText.propTypes,
  style: RNText.propTypes.style,
  typography: PropTypes.oneOf(typographiesKeys)
};

Text.defaultProps = {
  style: {},
  typography: 'title1'
};

export default function Text({ color, children, typography, style, ...attributes }) {
  return (
    <RNText style={StyleSheet.flatten([typographies[typography], style])} {...attributes}>
      {children}
    </RNText>
  );
}
