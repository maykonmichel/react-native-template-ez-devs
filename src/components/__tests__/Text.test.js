import React from 'react';
import { shallow } from 'enzyme';
import { forEach } from 'lodash';

import Text from '../Text';
import typographies from '../../configs/typographies';

describe('Testing Text component', () => {
  const text = shallow(<Text>Test</Text>);
  it('renders as expected', () => {
    expect(text).toMatchSnapshot();
  });

  forEach(typographies, (typography, key) => {
    it(`renders ${key} as expected`, () => {
      text.setProps({ typography });
      expect(text).toMatchSnapshot();
    });
  });
});
