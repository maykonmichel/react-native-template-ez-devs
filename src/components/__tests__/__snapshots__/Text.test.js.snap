// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Text component renders as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders as expected 2`] = `ShallowWrapper {}`;

exports[`Testing Text component renders body as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders callout as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders caption1 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders caption2 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders display1 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders display2 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders display3 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders display4 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders display5 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders footnote as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders headline as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders subhead as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders title1 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders title2 as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Text component renders title3 as expected 1`] = `ShallowWrapper {}`;
